skbio/stats/distance/_regression.py

```diff
@@ -0,0 +1,170 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2013--, scikit-bio development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+from __future__ import absolute_import, division, print_function
+
+import six
+import numpy as np
+from skbio.util._decorator import experimental
+from skbio.util._misc import check_random_state
+from skbio.stats.distance._mantel import _order_dms
+
+from skbio.stats.distance import DistanceMatrix
+
+@experimental(as_of="0.4.0")
+def linregress(y, *args, **kwargs):
```

Comment:
1. Statsmodels assumes normality, that that won't work out of the box.  However, if we _could_ perform permutations on that - but this will only work if statsmodels become a dependency.  This module is also pretty nice since it includes quite a few result statistics
2. Scikit-learn has the same problem with statsmodels, but worse since this will mean more dependencies
3. The scipy solution can be integrated into the solution that I have right now.  It could speed up the calculations of the coefficients.  But this will probably need profiling